src/test/java/com/pon02/Assignment10/integrationtest/CarTypeIntegrationTest.java

```diff
@@ -97,4 +102,94 @@ public class CarTypeIntegrationTest {
                 """, response, new CustomComparator(JSONCompareMode.STRICT,
                 new Customization("timestamp", ((o1, o2) -> true))));
     }
+
+    // POSTメソッドで正しくリクエストした時に、カータイプが登録できステータスコード201とメッセージが返されること
+    @Test
+    @DataSet(cleanBefore = true, cleanAfter = true, value = "datasets/car_types/car_types.yml")
+    @ExpectedDataSet(value = "datasets/car_types/insert_car_type.yml")
+    @Transactional
+    void カータイプが登録できること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.post("/car-types")
+                        .contentType("application/json")
+                        .content("""
+                                {
+                                    "carTypeName": "ハイエース9人乗り",
+                                    "capacity": 9
+                                }
+                                """))
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andExpect(MockMvcResultMatchers.header().exists("Location"))
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+        JSONAssert.assertEquals("""
+                {
+                    "message": "CarType created"
+                }
+                """, response, true);
+    }
+
+    // POSTメソッドでリクエストのcarTypeNameが空文字や50文字を超える場合、または登録済みの車種名を入力された時に、
+    // ステータスコード400とエラーメッセージが返されること
+    @Transactional
+    @ParameterizedTest
+    @MethodSource("provideStringsForValidation")
+    void 新規登録時carTypeNameが空文字または50文字以上または登録済みの車種名の場合400エラーが返されること(String str,String expectedMessage) throws Exception {
```

Comment:
今のテスト名だとMethodSourceに含めるパラメータが増えるたびにテスト名の変更も必要となりメンテナンスコストが高いですね。
```suggestion
    void 新規登録時carTypeNameが不正な値の場合400エラーが返されること(String str,String expectedMessage) throws Exception {
```
とかがよいかと！

Reply:
ご指摘ありがとうございます！
テスト名がどんどん長くなり微妙だなと感じていました。
テスト名修正とともにGitHubActionsが通らなくなったので@DataSetも追加しました。
297be10b93f451b299c96c534447771f4e8d57c6  
83e0efd74e5bb4a92ef0b5b16fd57e62963fa001

Reply:
テスト名変えてci通らなかったの謎ですね。
dataset追加したら通るのはOKなのですがなんで、テスト名変えたらfailするようになったのかを一応原因調べておいてほしいです。